akka-javasdk/src/main/scala/akka/javasdk/impl/timer/TimerSchedulerImpl.scala

```diff
@@ -23,21 +21,22 @@ import scala.jdk.DurationConverters.JavaDurationOps
 private[akka] final class TimerSchedulerImpl(val timerClient: akka.runtime.sdk.spi.TimerClient, val metadata: Metadata)
     extends TimerScheduler {
 
-  override def startSingleTimer[I, O](
-      name: String,
-      delay: Duration,
-      deferredCall: DeferredCall[I, O]): CompletionStage[Done] =
+  override def startSingleTimer[I, O](name: String, delay: Duration, deferredCall: DeferredCall[I, O]): Unit =
     startSingleTimer(name, delay, 0, deferredCall)
 
   override def startSingleTimer[I, O](
       name: String,
       delay: Duration,
       maxRetries: Int,
-      deferredCall: DeferredCall[I, O]): CompletionStage[Done] = {
+      deferredCall: DeferredCall[I, O]): Unit = {
 
     deferredCall match {
       case embeddedDeferredCall: DeferredCallImpl[I, O] =>
-        timerClient.startSingleTimer(name, delay.toScala, maxRetries, embeddedDeferredCall.deferredRequest()).asJava
+        timerClient
+          .startSingleTimer(name, delay.toScala, maxRetries, embeddedDeferredCall.deferredRequest())
+          .asJava
+          .toCompletableFuture
+          .get() // timeout handled by runtime
```

Comment:
I should unbox exceptions here like in the component client

Reply:
Added in [2ee912a](https://github.com/akka/akka-sdk/pull/348/commits/2ee912ae08c77644b5e5aac23787c1a6e638f7d3)